samtranslator/validator/sam_schema/definitions/api.json

```diff
@@ -0,0 +1,789 @@
+                ]
+              },
+              "CacheClusterSize": {
+                "type": [
+                  "string",
+                  "intrinsic"
+                ]
+              },
+              "CanarySetting": {
+                "$ref": "file:cfn.json#definitions/AWS::ApiGateway::Stage.CanarySetting"
+              },
+              "Cors": {
+                "if": {
+                  "type": "object"
+                },
+                "then": {
+                  "$ref": "#definitions/AWS::Serverless::Api.CorsConfiguration"
+                },
+                "type": [
+                  "string",
+                  "object",
+                  "intrinsic"
+                ]
+              },
+              "DefinitionBody": {
+                "type": [
+                  "string",
+                  "object",
+                  "intrinsic"
+                ]
+              },
+              "DefinitionUri": {
+                "type": [
+                  "string",
+                  "object"
+                ],
+                "$ref": "#definitions/AWS::Serverless::Api.S3Location"
+              },
+              "Description": {
+                "type": [
+                  "string",
+                  "intrinsic"
+                ]
+              },
+              "Domain": {
+                "$ref": "#definitions/AWS::Serverless::Api.DomainConfiguration"
+              },
+              "EndpointConfiguration": {
+                "$ref": "#definitions/AWS::Serverless::Api.EndpointConfiguration"
+              },
+              "GatewayResponses": {
+                "$ref": "file:common.json#definitions/GatewayResponses",
+                "references": [
+                  "https://docs.aws.amazon.com/apigateway/api-reference/resource/gateway-response/"
+                ]
+              },
+              "MethodSettings": {
+                "if": {
+                  "type": "array"
+                },
+                "then": {
+                  "items": {
+                    "$ref": "file:cfn.json#definitions/AWS::ApiGateway::Stage.MethodSetting"
+                  }
+                },
+                "type": [
+                  "array",
+                  "intrinsic"
+                ]
+              },
+              "MinimumCompressionSize": {
+                "type": [
+                  "integer",
+                  "intrinsic",
+                  "null"
+                ],
+                "maximum": 10485760,
+                "minimum": 0
+              },
+              "Models": {
+                "additionalProperties": {
+                  "type": "object"
+                },
+                "type": "object"
+              },
+              "Name": {
+                "type": [
+                  "string",
+                  "intrinsic"
+                ]
+              },
+              "OpenApiVersion": {
+                "type": "string",
+                "pattern": "^[2-3](\\.\\d)(\\.\\d)?$",
```

Comment:
Should this be `^(2\.0|3\.\d\.\d)?$` to only match 2.0 and 3.x.x instead of 2.x.x and 3.x.x?